### PR TITLE
Update comment to resolve ambiguity

### PR DESCRIPTION
**Pull request by mahmudsudo**
The documentation is incorrect as the determinant for lower or upper case is the 5th bit not the 6th bit as written in the codebase.
An excerpt from the book  the art of assembly language lays credence to this  :
"Upper case characters always contain a zero in bit five; lower case alphabetic characters always contain a one in bit five. You
can use this fact to quickly convert between upper and lower case. If you have an upper
case character you can force it to lower case by setting bit five to one."

### CODE REVIEW

**Comment by slanterns** [comment]
Simply wrong. 'O' = 79 = 0b010**0**1111, 'P' = 80 = 0b010**1**0000. Don't rely on the book without actually checking it.

**Comment by mahmudsudo** [comment]
E = 0 1 0 0 0 1 0 1
e = 0 1 1 0 0 1 0 1
The bit that determines the case of an ascii letter is the FIFTH BIT counted from the low order bit , this is further explained in the example given above

**Comment by mahmudsudo** [comment]
Incase we are still not clear ,the low order bit is the rightmost bit ,it has a place value of zero , hence called the zero bit , any bit number is counted from the zero bit as origin .

**Comment by slanterns** [comment]
I get what makes you confused. However I just found most people will call it the 6th bit by https://www.google.com/search?q=ascii+capital+bit+xor.

**Comment by mahmudsudo** [comment]
> Simply wrong. 'O' = 79 = 0b010**0**1111, 'P' = 80 = 0b010**1**0000. Don't rely on the book without actually checking it. (And if you read the code, it's also `const ASCII_CASE_MASK: u8 = 0b0010_0000` masking the 6th bit.)

nothing i said is actually wrong as pointed out by you , 
O = 01**0**01111
o= 01**1**01111
as shown the fifth bit (with the Low order bit as zero bit ) is the distinguishing factor not 6th bit )

**Comment by mahmudsudo** [comment]
> I get what makes you confused. However I just found most people will call it the 6th bit by https://www.google.com/search?q=ascii+capital+bit+xor.

The usage by people doesnt make the counting standard 
An 8 bit value is numbered X_0_ -- X_7_  even though there are 8 bits , the academia uses the fifth bit as the distinguisher between lowercase and uppercase .

**Comment by mahmudsudo** [comment]
> > Simply wrong. 'O' = 79 = 0b010**0**1111, 'P' = 80 = 0b010**1**0000. Don't rely on the book without actually checking it. (And if you read the code, it's also `const ASCII_CASE_MASK: u8 = 0b0010_0000` masking the 6th bit.)
> 
> nothing i said is actually wrong as pointed out by you , O = 01**0**01111 o= 01**1**01111 as shown the fifth bit (with the Low order bit as zero bit ) is the distinguishing factor not 6th bit )

https://www.geeksforgeeks.org/program-toggle-characters-string/

**Comment by Teapot4195** [comment]
Honestly we shouldn't be arguing over whether to call it 5th or 6th bit, both are equally confusing unless a labeled byte diagram is provided both 5 and 6 are valid. If you really wanted to know which bit was being masked a quick google search would tell you which bit is being masked

**Comment by slanterns** [comment]
> The usage by people doesnt make the counting standard

If you like standard, in [ISO-IR-6: ASCII Graphic character set](https://itscj.ipsj.or.jp/ir/006.pdf) or [ANSI INCITS 4-1986](http://unicode.org/L2/L2006/06388-review-incits4.pdf) they are called b1..b7 (since b8 is not used). I think arguing for things like that is just not so meaningful.

File: library/core/src/num/mod.rs [hunk 1]
```diff
@@ -478,7 +478,7 @@ impl isize {
     }
 }
 
-/// If the 6th bit is set ascii is lower case.
+/// If the 5th bit is set ascii is lower case.
```
Comment (Mark-Simulacrum) [inline library/core/src/num/mod.rs:481]:
```suggestion
/// If the bit selected by this mask is set, ascii is lower case.
```

**Comment by Mark-Simulacrum** [comment]
r=me with suggestion or equivalent text applied and squashed into one commit. Otherwise I'm not inclined to change one number to the other; both seem equally not helpful to me.

**Comment by Teapot4195** [comment]
Squash please

**Comment by Dylan-DPC** [comment]
@mahmudsudo any updates on this?

**Comment by mahmudsudo** [comment]
I have resolved the suggested change to the commit

**Comment by Mark-Simulacrum** [comment]
This still needs to be squashed.

@rustbot author

**Comment by oskgo** [comment]
@mahmudsudo

Ping from triage: I'm closing this due to inactivity, Please reopen when you are ready to continue with this.
Note: if you are going to continue please open the PR BEFORE you push to it, else you won't be able to reopen - this is a quirk of github.
Thanks for your contribution.

@rustbot label: +S-inactive

**Comment by chansuke** [comment]
I'd like to help move this PR forward

**Pull request closed without merging**